libtiledbsoma/src/soma/soma_array.cc

```diff
@@ -1175,32 +1175,62 @@ uint64_t SOMAArray::nnz_slow() {
     return total_cell_num;
 }
 
+bool SOMAArray::_dims_are_int64() {
+    ArraySchema schema = arr_->schema();
+    Domain domain = schema.domain();
+    for (auto dimension : domain.dimensions()) {
+        if (dimension.type() != TILEDB_INT64) {
+            return false;
+        }
+    }
+    return true;
+}
+
+void SOMAArray::_assert_dims_are_int64() {
+    if (!_dims_are_int64()) {
+        throw TileDBSOMAError(
+            "[SOMAArray] internal coding error: expected all dims to be int64");
```

Comment:
Should this be an `assert` instead of throwing an exception then?

Reply:
I want it to throw ... open to a better method  name :)

Reply:
`_check_dims_are_int64` ✨